### ISSUE TITLE
EXT3 unsupported optional features (2c0)

### ISSUE THREAD

**Issue by kcgthb** [apptainer/singularity#50]
Hi!

I just installed v2.0 and created a Debian container image (on a Debian machine), which works great:

```
debian $ singularity exec container.img cat /etc/debian_version
8.4
```

Now, I transferred that container image and try to "run" it on a Centos machine, and there seem to be a problem with the way the filesystem image is formatted:

```
centos $ singularity exec container.img cat /etc/debian_version
ERROR: Failed to mount (ro) '/dev/loop0' at '/tmp/.singularity-215845.2032753612.144117941686781743/mnt': Invalid argument
```

And the kernel logs on the Centos host say:

```
JBD: Ignoring recovery information on journal
JBD: Unrecognised features on journal
EXT4-fs (loop0): error loading journal
JBD: Ignoring recovery information on journal
JBD: Unrecognised features on journal
EXT4-fs (loop0): error loading journal
EXT3-fs (loop0): error: couldn't mount because of unsupported optional features (2c0)
```

Any idea what's wrong here?

Thanks!

**Comment by gmkurtzer** [comment]
Hrmm, I'm not sure actually. It looks like the mkfs.ext4 on the image is not portable between Debian and Centos! Very odd, I haven't seen that before. 

Do me a favor please, and try the following... Edit libexec/cli/create.exec and comment out the mkfs.ext4 bits like this:

```
#if ! MKFS_PATH=`singularity_which mkfs.ext4`; then
    if ! MKFS_PATH=`singularity_which mkfs.ext3`; then
        message ERROR "Could not locate program: mkfs.ext3/4\n"
        exit 255
    fi
#fi
```

This way we can force it to use ext3 instead. And then recreate an image, and rebootstrap and try to bring that over to the Centos system.

Thanks!

**Comment by kcgthb** [comment]
Sure, thanks for looking at it. 

I commented the ext4 part, then re-created and boostraped the Debian image, re-imported it on the Centos machine, and now it works.

Which made me wonder if my initial ext4 image had been corrupted somehow, and only worked on the Debian machine because of some sort of caching. So I re-enabled mkfs.ext4 in `create.exec`, and created a new container image. But still no dice, same "unsupported optional features (2c0)" error when mounting the image on Centos.

**Comment by kcgthb** [comment]
And for the sake of completeness:

```
debian # mkfs.ext4 -V
mke2fs 1.43 (17-May-2016)
        Using EXT2FS Library version 1.43
```

```
centos # mkfs.ext4 -V
mke2fs 1.41.12 (17-May-2010)
        Using EXT2FS Library version 1.41.12
```

**Comment by gmkurtzer** [comment]
I think for the sake or portability, I will default to ext3 and offer an option to use ext4. This has been committed in [master 9f27c61] and will be available in v2.1.

Thanks!

**Comment by kcgthb** [comment]
Sounds good, thank you!
